### PR TITLE
Changed protobuf source code id to the latest one

### PR DESCRIPTION
**Pull request by kadamvandita**
Changed protobuf source code id to the latest one.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: go/private/repositories.bzl [hunk 1]
```diff
@@ -98,8 +98,8 @@ def go_rules_dependencies():
   _maybe(native.http_archive,
       name = "com_google_protobuf",
       # v3.5.0, latest as of 2017-11-24
-      url = "https://codeload.github.com/google/protobuf/zip/2761122b810fe8861004ae785cc3ab39f384d342",
-      strip_prefix = "protobuf-2761122b810fe8861004ae785cc3ab39f384d342",
+      url = "https://codeload.github.com/google/protobuf/zip/642e1ac635f2563b4a14c255374f02645ae85dac",
```
Comment (jayconrod) [inline go/private/repositories.bzl]: I'd prefer to stick to official releases. Let's use v3.5.1.1 instead.

Also, please update the comment above.

Comment (jayconrod) [reply go/private/repositories.bzl]: I switched this to v3.5.1. v3.5.1.1 is not marked as an official release and has no notes or downloads. Not sure what the story is there.

**Comment by jayconrod** [comment]
Jenkins test this please

**Comment by kadamvandita** [comment]
i wanted the code commit which has support for  Atomic64 Release_CompareAndSwap in the file  protobuf/src/google/protobuf/stubs/atomicops_internals_generic_gcc.h. 
The code is not yet supported in any stab;e branch, it is present in master or code with commit id: 642e1ac635f2563b4a14c255374f02645ae85dac. Can that be done?

**Comment by jayconrod** [comment]
@kadamvandita I'd prefer not to depend on a commit which is not tagged as an official release. I don't know how stable they are between releases.

Could you override this repo in your WORKSPACE file instead? Just add an `http_archive` rule like the one modified in this PR (without `_maybe`) before the call to `go_rules_dependencies`.

**Pull request merged**